services/debugger/destination/eventDeliveryStatusUploader_test.go

```diff
@@ -222,12 +224,13 @@ var _ = Describe("eventDeliveryStatusUploader", func() {
 					}()
 				})
 			tFunc := c.asyncHelper.ExpectAndNotifyCallback()
-			mockCall.Do(func(channel chan pubsub.DataEvent, topic backendconfig.Topic) { tFunc() }).Return().Times(1)
+			mockCall.Do(func(channel chan pubsub.DataEvent, topic backendconfig.Topic) {
+				tFunc()
+			}).Return().Times(1)
 
-			time.Sleep(1 * time.Second)
+			c.asyncHelper.WaitWithTimeout(5 * time.Second)
```

Comment:
5s for a unit tests seems like a very long timeout, can you give a bit more context as to why we would need 5s?

Reply:
All the tests in the service/debugger directory run in 2s locally for me. So this is mostly like a theoretical maximum. We could lower to 1 second if that helps.